server/storage/types.go

```diff
@@ -4,7 +4,29 @@ import "github.com/docker/notary/tuf/data"
 
 // MetaUpdate packages up the fields required to update a TUF record
 type MetaUpdate struct {
-	Role    data.RoleName
-	Version int
-	Data    []byte
+	Role     data.RoleName
+	Version  int
+	Data     []byte
+	Channels []*Channel
```

Comment:
(The issue with the in-memory implementation is the answer to your other comment as well, for some reason GH won't let me reply to it)